cEP-0016.md

```diff
@@ -0,0 +1,332 @@
+   CI build that is with respect to modified files and whether the corresponding
+   CI engine allows skipping build or not. An appropriate message is returned if
+   skipping build is not supported.
+   - If the commit type from the output of `VCSCommitBear` indicates that the
+     commit skips CI build, inspection is done inside `SkipCIBear` otherwise
+     return.
+   - Files modified in the commit should include files that are not covered by
+     `.coafile` or tests. If this is not the case, an error is raised to exclude
+     `[ci skip]` or `[skip ci]` from the commit message.
+6. For merge commits:
+   - If the commit type from the output of `VCSCommitBear` indicates that the
+     commit is a `git merge` commit, inspection is done in `MergeCommitBear`
+     otherwise return.
+   - `MergeCommitBear` will be used by find if the commit tries to merge master
+     or any other branch onto the pull request, message is displayed to perform
+     a `git rebase` instead.
+7. For revert commits:
+   - If the commit type from the output of `VCSCommitBear` indicates that the
+     commit is a `git revert` commit, inspection is done in `RevertCommitBear`
+     otherwise return.
+   - In `RevertCommitBear`, the patches of both commits are compared to
+     ensure that they are in fact reverse of each other. Minor differences are
+     allowed.
+   - In case of total mismatch, a note in the commit message highlighting that
+     it is not a clean revert is expected.
+   - A message suggesting improvements in the commit message is displayed.
+
+## Code Prototypes
+
+A prototype for identifying GitHub Pull Request temporary merge commits:
+
+```python
+    if len(shortlog) > shortlog_length:
+        github_pr_merge_commit_regex = re.compile(r'^Merge [0-9a-f]+ into [0-9a-f]+$|....')
+
+        match = re.fullmatch(github_pr_merge_commit_regex, shortlog)
+        if match:
+            # ignore this commit from any further inspection
+            # inspect the unmerged parent of this commit instead
+```
+
+A prototype of `VCSCommitBear` :
+
+```python
+    # import statements
+
+    class CommitResult(HiddenResult):
+        def __init__(self, origin, commit_info, message):
+            Result.__init__(self, origin, commit_info)
+            self.commit_info = commit_info
+            self.message = message
+
+
+    class VCSCommitBear(GlobalBear):
+        __metaclass__ = abc.ABCMeta
+        AUTHORS = {'The coala developers'}
+        AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+        LICENSE = 'AGPL-3.0'
+
+        @abc.abstractmethod
+        def get_head_commit(self):
+            """
+            Return the commit message from the head commit
+            """
+
+        def analyze_git_commit(self, stdout):
+            commit_info = dict()
+
+            commit_sha = run_shell_command('git rev-parse HEAD')[0].strip('\n')
+
+            commit_type = []
+
+            stdout = stdout.strip('\n')
```

Comment:
I assume you are missing a line here, where stdout should be the commit message?

IMO, this bear should store the raw commit message in the Result object for the other bears to use.